sdk/python/ydk/services/meta_service.py

```diff
@@ -220,8 +225,13 @@ def inject_imeta_helper(entity, deviation_table, parent=None):
         elif isinstance(value, READ) or isinstance(value, DELETE):
             continue
         elif member.mtype == REFERENCE_LIST:
-            for v in value:
-                inject_imeta_helper(v, deviation_table, entity.i_meta)
+            try:
+                for v in value:
+                    inject_imeta_helper(v, deviation_table, entity.i_meta)
+            except TypeError:
+                raise YPYModelError("Assigned object to YList {},"
+                                    " use list method instead."
```

Comment:
OK. I think this should be YPYServiceError. How about the below:

YPYServiceError(' Assigned object to YLeafList. Use list append() instead.'?)